tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -1810,6 +1812,76 @@ int BPF_KPROBE(trace_security_inode_unlink)
     return 0;
 }
 
+SEC("kprobe/commit_creds")
+int BPF_KPROBE(trace_commit_creds)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    u8 argnum = 0;
+    context_t context = init_and_save_context(ctx, submit_p, COMMIT_CREDS, 2 /*argnum*/, 0 /*ret*/);
+
+    struct cred *new = (struct cred *)PT_REGS_PARM1(ctx);
+
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    struct cred *old = (struct cred *)READ_KERN(task->real_cred);
+
+    kuid_t old_euid = READ_KERN(old->euid);
+    kuid_t new_euid = READ_KERN(new->euid);
+    kgid_t old_egid = READ_KERN(old->egid);
+    kgid_t new_egid = READ_KERN(new->egid);
+    kuid_t old_fsuid = READ_KERN(old->fsuid);
+    kuid_t new_fsuid = READ_KERN(new->fsuid);
+    kernel_cap_t old_cap_eff = READ_KERN(old->cap_effective);
+    kernel_cap_t new_cap_eff = READ_KERN(new->cap_effective);
+
+    // Currently (2021), there are ~40 capabilities in the Linux kernel which are stored in a 2 dimentional u32 array
+    // This might change in the (not so near) future as more capabilities will be added.
+    // For now, we use u64 to store this array in one piece
```

Comment:
if you want, you could check if `_KERNEL_CAPABILITY_U32S == 2` to verify

Reply:
It will take a few more years until this will be changed to length of 3 (more than 20 capabilities will have to be added first), and we do not support kernel version with cap length of 1 - so I think it is ok  to skip such test for now